<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Controlwindow</class>
 <widget class="QMainWindow" name="Controlwindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1235</width>
    <height>600</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1235</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(0, 0, 0);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="Control">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>0</y>
      <width>1900</width>
      <height>1200</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);

font: 12pt &quot;Times new roman&quot;;
color: white
</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="Gain">
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);
color: white;</string>
     </property>
     <attribute name="title">
      <string>Gain Set</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>40</y>
        <width>501</width>
        <height>381</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(30, 30, 30);</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QDoubleSpinBox" name="RR_r_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>160</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_r_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>100</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_r_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>160</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;RR_r&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>20</y>
         <width>91</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; text-decoration: underline; color:#ffffff;&quot;&gt;P_gain&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_r_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>70</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_r_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>160</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_r_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>130</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;RL_r&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;FL_r&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_r_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>71</y>
         <width>81</width>
         <height>20</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_r_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>130</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_r_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>100</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_r_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>100</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;FR_r&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>20</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; text-decoration: underline; color:#ffffff;&quot;&gt;D_gain&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_r_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>70</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_r_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>130</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>20</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; text-decoration: underline; color:#ffffff;&quot;&gt;I_gain&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>20</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; text-decoration: underline; color:#ffffff;&quot;&gt;Cut_Off&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#ffffff;&quot;&gt;&lt;br/&gt;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_r_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>70</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_r_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>100</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_r_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>160</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_r_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>130</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>230</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;FR_th&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;FL_th&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>290</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;RR_th&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>260</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;RL_th&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_th_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>200</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_th_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>230</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_th_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>290</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_th_posPgain">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>260</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_th_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>200</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_th_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>260</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_th_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>290</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_th_posIgain">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>230</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_th_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>200</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_th_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>230</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_th_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>290</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_th_posDgain">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>260</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RR_th_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>290</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FR_th_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>230</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="RL_th_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>260</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="FL_th_posD_cutoff">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>200</y>
         <width>81</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(255, 255, 255);
text-decoration: underline;</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>10</y>
        <width>131</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#ffffff;&quot;&gt;Position PID&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Set">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>40</y>
        <width>151</width>
        <height>71</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">
background-color: rgb(200, 0, 0);</string>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Graph">
     <attribute name="title">
      <string>Graph</string>
     </attribute>
     <widget class="QCustomPlot" name="RL_pos_plot" native="true">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>50</y>
        <width>301</width>
        <height>201</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>20</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>RL_position</string>
      </property>
     </widget>
     <widget class="QCustomPlot" name="RLHIP_pos_plot" native="true">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>50</y>
        <width>291</width>
        <height>211</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>20</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>RLHIP_position</string>
      </property>
     </widget>
     <widget class="QCustomPlot" name="RLKNEE_pos_plot" native="true">
      <property name="geometry">
       <rect>
        <x>740</x>
        <y>50</y>
        <width>311</width>
        <height>211</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>840</x>
        <y>20</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>RLKNEE_position</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="EtherCAT">
     <property name="font">
      <font>
       <family>Times new roman</family>
       <pointsize>12</pointsize>
       <italic>false</italic>
       <bold>false</bold>
      </font>
     </property>
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;EtherCAT&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <attribute name="title">
      <string>EtherCAT</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
